ui/v2.5/src/components/Images/ImageWall.tsx

```diff
@@ -0,0 +1,191 @@
+import React, { MouseEvent, useMemo, useRef } from "react";
+import { Button, ButtonGroup } from "react-bootstrap";
+import { PerformerPopoverButton } from "../Shared/PerformerPopoverButton";
+import {
+  faBox,
+  faImages,
+  faPlus,
+  faTag,
+} from "@fortawesome/free-solid-svg-icons";
+import { Icon, TagLink, HoverPopover, SweatDrops } from "src/components/Shared";
+import * as GQL from "src/core/generated-graphql";
+import { objectTitle } from "src/core/files";
+import { Link } from "react-router-dom";
+
+interface IImageWallProps {
+  images: GQL.SlimImageDataFragment[];
+  onChangePage: (page: number) => void;
+  currentPage: number;
+  pageCount: number;
+  handleImageOpen: (index: number) => void;
+}
+
+export const ImageWall: React.FC<IImageWallProps> = ({
+  images,
+  handleImageOpen,
+}) => {
+  const thumbs = images.map((image, index) => {
+    function maybeRenderTagPopoverButton() {
+      if (image.tags.length <= 0) return;
+
+      const popoverContent = image.tags.map((tag) => (
+        <TagLink key={tag.id} tag={tag} tagType="image" />
+      ));
+
+      return (
+        <HoverPopover
+          className="tag-count"
+          placement="bottom"
+          content={popoverContent}
+        >
+          <Button className="minimal">
+            <Icon icon={faTag} />
+            <span>{image.tags.length}</span>
+          </Button>
+        </HoverPopover>
+      );
+    }
+
+    function maybeRenderPerformerPopoverButton() {
+      if (image.performers.length <= 0) return;
+
+      return <PerformerPopoverButton performers={image.performers} />;
+    }
+
+    function maybeRenderOCounter() {
+      if (image.o_counter) {
+        return (
+          <div className="o-count">
+            <Button className="minimal">
+              <span className="fa-icon">
+                <SweatDrops />
+              </span>
+              <span>{image.o_counter}</span>
+            </Button>
+          </div>
+        );
+      }
+    }
+
+    function maybeRenderGallery() {
+      if (image.galleries.length <= 0) return;
+
+      const popoverContent = image.galleries.map((gallery) => (
+        <TagLink key={gallery.id} gallery={gallery} />
+      ));
+
+      return (
+        <HoverPopover
+          className="gallery-count"
+          placement="bottom"
+          content={popoverContent}
+        >
+          <Button className="minimal">
+            <Icon icon={faImages} />
+            <span>{image.galleries.length}</span>
+          </Button>
+        </HoverPopover>
+      );
+    }
+
+    function maybeRenderOrganized() {
+      if (image.organized) {
+        return (
+          <div className="organized">
+            <Button className="minimal">
+              <Icon icon={faBox} />
+            </Button>
+          </div>
+        );
+      }
+    }
+
+    function renderEdit() {
```

Comment:
Adding a "+" icon/button here so that you can still see the bottom bar and access editing if no other element present in it